lib/sogou/search/api/service/keyword.rb

```diff
@@ -9,18 +9,30 @@ def initialize
             super('http://api.agent.sogou.com:8080/sem/sms/v1/CpcService?wsdl')
           end
 
-          def get_cpc_by_cpc_grp_id(grp_ids, valid_keyword_only=true)
-            resp = client.call(:get_cpc_by_cpc_grp_id,
-                               message: { cpc_grp_ids: grp_ids, get_temp: valid_keyword_only ? 0 : 1 })
-            puts resp.header
-            resp.body
+          def get_cpc_by_cpc_grp_id(grp_ids, valid_keyword_only = true, options = {}, &block)
+            command = make_command(:get_cpc_by_cpc_grp_id,
+                                   params: {
+                                     cpc_grp_ids: grp_ids,
+                                     get_temp: valid_keyword_only?(valid_keyword_only)
+                                   },
+                                   options: options)
+            execute_command(command, &block)
           end
 
-          def get_cpc_id_by_cpc_grp_id(grp_ids, valid_keyword_only=true)
-            resp = client.call(:get_cpc_id_by_cpc_grp_id,
-                               message: { cpc_grp_ids: grp_ids, get_temp: valid_keyword_only ? 0 : 1  })
-            puts resp.header
-            resp.body
+          def get_cpc_id_by_cpc_grp_id(grp_ids, valid_keyword_only = true, options = {}, &block)
+            command = make_command(:get_cpc_id_by_cpc_grp_id,
+                                   params: {
+                                     cpc_grp_ids: grp_ids,
+                                     get_temp: valid_keyword_only?(valid_keyword_only)
+                                   },
+                                   options: options)
+            execute_command(command, &block)
+          end
+
+          private
+
+          def valid_keyword_only?(only)
```

Comment:
How come a method with a `?` signature would return `0` or `1`?

Reply:
:) it doesn't seem right